src/core/modules/impl/TagsParser.ts

```diff
@@ -0,0 +1,48 @@
+import { GQLEdgeInterface, GQLTagInterface, LoggerFactory, SmartWeaveTags } from '@smartweave';
+
+/**
+ * A class that is responsible for retrieving "input" tag from the interaction transaction.
+ * Two tags formats are allowed:
+ * 1. "multiple interactions in one tx" format - where "Input" tag MUST be next to the "Contract" tag
+ *    See more at https://github.com/ArweaveTeam/SmartWeave/pull/51
+ * 2. "traditional" format - one interaction per one transaction - where tags order does not matter.
+ *
+ * More on Discord: https://discord.com/channels/357957786904166400/756557551234973696/885388585023463424
+ */
+export class TagsParser {
+  private readonly logger = LoggerFactory.INST.create('TagsParser');
+
+  getInputTag(interactionTransaction: GQLEdgeInterface, contractTxId: string): GQLTagInterface {
+    // this is the part to retain compatibility with https://github.com/ArweaveTeam/SmartWeave/pull/51
+    if (TagsParser.hasMultipleInteractions(interactionTransaction)) {
+      this.logger.debug('Interaction transaction is using multiple input tx tag format.');
+      const contractTagIndex = interactionTransaction.node.tags.findIndex(
+        (tag) => tag.name === SmartWeaveTags.CONTRACT_TX_ID && tag.value === contractTxId
```

Comment:
Shouldn't this be:
```js
interactionTransaction.node.tags.findIndex((tag) => tag.name === 'Contract' && tag.value === contractTxId);
```

Unless `SmartWeaveTags.CONTRACT_TX_ID === 'Contract'` This might be a bit confusing if that's the case?

Reply:
Confirmed `CONTRACT_TX_ID` is equal to `'Contract'`.

Reply:
SmartWeaveTags is an enum created to keep all the tags defined and used by the protocol in the one place.